src/solvers/dgsem_tree/indicators.jl

```diff
@@ -102,6 +102,35 @@ function Base.show(io::IO, ::MIME"text/plain", indicator::IndicatorHennemannGass
 end
 
 
+function (indicator_hg::IndicatorHennemannGassner)(u, mesh, equations, dg::DGSEM, cache;
+                                                   kwargs...)
+  @unpack alpha_smooth = indicator_hg
+  @unpack alpha, alpha_tmp = indicator_hg.cache
+  # TODO: Taal refactor, when to `resize!` stuff changed possibly by AMR?
+  #       Shall we implement `resize!(semi::AbstractSemidiscretization, new_size)`
+  #       or just `resize!` whenever we call the relevant methods as we do now?
+  resize!(alpha, nelements(dg, cache))
+  if alpha_smooth
+    resize!(alpha_tmp, nelements(dg, cache))
+  end
+
+  # magic parameters
+  threshold = 0.5 * 10^(-1.8 * (nnodes(dg))^0.25)
+  parameter_s = log((1 - 0.0001)/0.0001)
+
+  @threaded for element in eachelement(dg, cache)
+    # This is dispatched by mesh dimension.
+    calc_element_alpha_hg!(indicator_hg, threshold, parameter_s, u, element,
+                           mesh, equations, dg, cache)
+  end
```

Comment:
Could you please add a comment here, too, exapliniexplainingng the need for a function barrier?

Reply:
Should I remove the one at the function barrier then? Or keep both comments?

Reply:
Keeping both would be great - a bit of redundancy explaining something like this is always good